Акимов Денис

maBna opened the pull request.

maBna committed:
Аким

maBna committed:
Акимов

Единственно, не понимаю, почему filterEqual возвращает всю коллекцию

i4got10 commented on lego.js:
```diff
@@ -1,11 +1,144 @@
-// Оператор reverse, который переворачивает коллекцию
+module.exports.select = function () {
+	var whatNeed = [false, false, false, false, false, false];
+	var properties = ['age', 'gender', 'name', 'email', 'phone','favoriteFruit'];
```
можно получить через Object.keys

i4got10 commented on lego.js:
```diff
@@ -1,11 +1,144 @@
+	var properties = ['age', 'gender', 'name', 'email', 'phone','favoriteFruit'];
+	for (var i = 0; i < arguments.length; i++)
+	{	
+		for (var j = 0; j < 6; j++)
```
что за 6

maBna committed:
Update lego.js

maBna commented on lego.js:
```diff
@@ -1,11 +1,153 @@
 
 };
 
-// Оператор reverse, который переворачивает коллекцию
```
Поправил весь селект, теперь с помощью Object.keys получаю ключи
Число 6 было количеством ключей, тоже поменял условие, теперь это длина соответствующего массива с ключами

maBna committed:
Update lego.js

maBna commented on lego.js:
```diff
@@ -1,11 +1,152 @@
+		return collection;
+	};
+};
+
```
теперь работает
Я не понял, чем ему не по нраву было возвращать измененную коллекцию

i4got10 commented on lego.js:
```diff
@@ -1,11 +1,152 @@
+	{
+		for (var i = 0; i < fruits.length; i++){
+			for (var j = 0; j < collection.length; j++){
+					if (collection[j] != null)
```
что за условие? Для чего оно?

i4got10 commented on lego.js:
```diff
@@ -1,11 +1,152 @@
-// Оператор reverse, который переворачивает коллекцию
+module.exports.select = function () {
+	var conditions = [];
+	var whatNeed = [false, false, false, false, false, false];
```
а что за 6 false?

maBna replied:
Это был массив из нужных нам ключей объекта, т.е. true - нужен, false - нет

i4got10 commented:
.filterIn работает неправильно сейчас.
Искал вот так

`lego.query(collection, lego.filterIn('favoriteFruit', ['Картофель', 'Яблоко']))`
должно быть 51, вернуло 52

`lego.query(collection, lego.filterIn('favoriteFruit', []))`
должно быть 0, получил 100

`lego.query(collection, lego.filterIn('unknownField', ['fistValue', 'secondValue']))`
должно быть 0, получил 100

maBna committed:
большие правки

Переделал limit, теперь если число отрицательное или больше длины коллекции, ничего не режем
Переделал filterEqual,filterIn с помощью indexOf, теперь вроде все отрабатывает правильно
Подправил select и sortBy

maBna committed:
опрятный вид

постарался поправить в соответствии с codestyle

i4got10 commented:
1) select не работает
2) limit неверно работает с нулем, и отрицательными числами

maBna committed:
change limit

maBna committed:
change limit and select

maBna commented:
Поменял select - теперь там два for'а и нет удаления свойств
limit - теперь при отрицательных и нулевых значениях выдает []

maBna commented:
Не на ту кнопку ткнул, прошу прощения

maBna commented:
А в селект мы можем отправить то, чего нет в ключах коллекции?
UPD. Сделал, чтоб не пытался добавлять левые свойства

maBna committed:
Update lego.js

i4got10 commented:
Принято! Теперь все работает, кроме доп. задания
